autogen.sh

```diff
@@ -1004,7 +1005,7 @@ if [ "$do_build_configure" = "yes" ] ; then
             # Patching ltmain.sh
             if [ -f $amdir/confdb/ltmain.sh ] ; then
                 echo_n "Patching ltmain.sh for compatibility with Intel compiler options... "
-                patch -N -s -l $amdir/confdb/ltmain.sh maint/patches/optional/confdb/intel-compiler.patch
+                patch -N -s -l $amdir/confdb/ltmain.sh maint/patches/optional/confdb/intel-compiler.patch && :
```

Comment:
So `&& :` seems to work, but why? I'm not familiar with this shell pattern.

Reply:
`-e` just ignores components with `&&` or `||`.  And `:` doesn't overwrite `$?`.

Reply:
I'm was mostly trying to understand what `:` does. Finally was able to figure out it is a shell builtin.